make/modules/java.desktop/lib/ClientLibraries.gmk

```diff
@@ -246,6 +246,7 @@ ifeq ($(ENABLE_HEADLESS_ONLY), false)
       DISABLED_WARNINGS_clang := deprecated-non-prototype, \
       DISABLED_WARNINGS_clang_dgif_lib.c := sign-compare, \
       DISABLED_WARNINGS_clang_gzwrite.c := format-nonliteral, \
+      DISABLED_WARNINGS_clang_png.c := unused-function, \
```

Comment:
This warning was disabled to fix the following build issue on linux and macOS.

```
open/src/java.desktop/share/native/libsplashscreen/libpng/png.c:1558:1: error: 'png_icc_profile_error' defined but not used [-Werror=unused-function]
[2025-03-12T23:57:44,575Z]  1558 | png_icc_profile_error(png_const_structrp png_ptr, png_const_charp name,
[2025-03-12T23:57:44,575Z]       | ^~~~~~~~~~~~~~~~~~~~~
```

Reply:
In libconfig.h Ln#120 the undef is commented out - /*#undef PNG_READ_iCCP_SUPPORTED*/.

@prrace  do we define it in libconfig.h and remove the added warning ? In Updating.txt , I see a note on libconfig.h, it is heavily modified for JDK and a define stmt can be added on need basis.

Reply:
@mrserb I see, what you meant is that the function `png_icc_profile_error` should be guarded with PNG_READ_iCCP_SUPPORTED instead of PNG_iCCP_SUPPORTED. Reported it to upstream - https://github.com/pnggroup/libpng/issues/668